### PR TITLE
Feat: introduce redis-json cachified adapter

### PR DESCRIPTION
**Pull request by tearingItUp786**
[Issue Link Describing the Request](https://github.com/Xiphe/cachified/issues/45)

Introduce the redis json like cache adapter. I was unable to find a `redis-json-mock` library for us to use so I opted to follow the pattern that exists in the current `adapters.ts` and created a `RedisJsonLikeCache`

I have a similar implementation on [my own site that you can reference.](https://github.com/tearingItUp786/blog/blob/main/app/utils/redis.server.ts#L48) 

Side note: sorry for the MIA -- got busy with life 🙇.

### CODE REVIEW

File: src/adapters.ts [hunk 1]
```diff
@@ -142,3 +142,57 @@ export function redisCacheAdapter(redisCache: RedisLikeCache): Cache {
...
+
+export function redisJsonCacheAdapter(
+  redisJsonCache: RedisJsonLikeCache,
+): Cache {
+  return {
+    name: redisJsonCache.name || 'RedisJson',
+    delete(key) {
+      return redisJsonCache.json.del(key);
+    },
+    async set(key: string, value: Record<string, any>) {
```
Comment (Xiphe) [inline src/adapters.ts:168]: We should be able to remove the explicit typing here and go with `set(key, value)` that way we have type-safety on things like `value?.metadata`

**Review by Xiphe — CHANGES_REQUESTED** [review]
Awesome! Thank you for the contribution! Can we get rid of the `any`?

**Comment by kentcdodds** [comment]
Hello! I'm taking over as the primary maintainer. One thing I'd like to do is keep adapters out of this package and instead have them be their own package and we can simply add a reference in the docs. That will help us avoid unnecessary breaking change releases due to adapters needing to change.

Would you be interested in making an adapter package with `@epic-web/cachified` as a peer dependency and then making a PR to link to your package?

**Comment by tearingItUp786** [comment]
Hey @kentcdodds  👋🏽! Sure; sounds like fun! Have never set up a package/deployed anything to NPM but no time like the present. For this new package, are you wanting to have it under the same namespace such as `@epic-web/cachified-adapters`?

**Comment by kentcdodds** [comment]
I think it'd be best if you publish it yourself. You can namespace it under your own username on npm or just publish it without a namespace. I don't think a collection of adapters would be good. Instead, each adapter can be its own package. You could call this one `cachified-redis-json` to be descriptive 😅

The nice thing about this is you get complete control over the package and don't need anyone's permission to make changes as you see fit.

**Comment by tearingItUp786** [comment]
Create [a PR for the link to the new package](https://github.com/epicweb-dev/cachified/pull/78)! Will close this one

**Pull request closed without merging**